---
title: TypeError
slug: Web/JavaScript/Reference/Global_Objects/TypeError
tags:
  - Error
  - JavaScript
  - Object
  - Reference
  - TypeError
translation_of: Web/JavaScript/Reference/Global_Objects/TypeError
---
<div>{{JSRef("Global_Objects", "Error", "EvalError,InternalError,RangeError,ReferenceError,SyntaxError,TypeError,URIError")}}</div>

<h2 id="Summary">Сводка</h2>

<p>Объект <code><strong>TypeError</strong></code> представляет ошибку, возникающую, когда значение имеет не ожидаемый тип.</p>

<h2 id="Syntax">Синтаксис</h2>

<pre class="syntaxbox"><code>new TypeError([<var>message</var>[, <var>fileName</var>[, <var>lineNumber</var>]]])</code></pre>

<h3 id="Parameters">Параметры</h3>

<dl>
 <dt><code>message</code></dt>
 <dd>Необязательный параметр. Человеко-читаемое описание ошибки.</dd>
 <dt><code>fileName</code> {{non-standard_inline}}</dt>
 <dd>Необязательный параметр. Имя файла, содержащего код, вызвавший исключение.</dd>
 <dt><code>lineNumber</code> {{non-standard_inline}}</dt>
 <dd>Необязательный параметр. Номер строки кода, вызвавшей исключение.</dd>
</dl>

<h2 id="Description">Описание</h2>

<p>Исключение <code>TypeError</code> выбрасывается, когда операнд или аргумент, переданный в функцию, не совместим с типом, ожидаемым оператором или функцией.</p>

<h2 id="Properties">Свойства</h2>

<dl>
 <dt>{{jsxref("TypeError.prototype")}}</dt>
 <dd>Позволяет добавлять свойства в объект <code>TypeError</code>.</dd>
</dl>

<h2 id="Methods">Методы</h2>

<p>Глобальный объект <code>TypeError</code> не содержит собственных методов, однако, он наследует некоторые методы из цепочки прототипов.</p>

<h2 id="TypeError_instances">Экземпляры объекта <code>TypeError</code></h2>

<h3 id="Properties_of_TypeError_instances">Свойства</h3>

<div>{{page('/ru/docs/Web/JavaScript/Reference/Global_Objects/TypeError/prototype', 'Properties')}}</div>

<h3 id="Methods_of_TypeError_instances">Методы</h3>

<div>{{page('/ru/docs/Web/JavaScript/Reference/Global_Objects/TypeError/prototype', 'Methods')}}</div>

<h2 id="Examples">Примеры</h2>

<h3 id="Example:_Catch_an_TypeError">Пример: перехват исключения <code>TypeError</code></h3>

<pre class="brush: js">try {
  null.f();
} catch (e) {
  console.log(e instanceof TypeError); // true
  console.log(e.message);              // "null has no properties" - null не имеет свойств
  console.log(e.name);                 // "TypeError"
  console.log(e.fileName);             // "Scratchpad/1"
  console.log(e.lineNumber);           // 2
  console.log(e.columnNumber);         // 2
  console.log(e.stack);                // "@Scratchpad/2:2:3\n"
}
</pre>

<h3 id="Example:_Create_an_TypeError">Пример: возбуждение исключения <code>TypeError</code></h3>

<pre class="brush: js">try {
  throw new TypeError('Привет', 'someFile.js', 10);
} catch (e) {
  console.log(e instanceof TypeError); // true
  console.log(e.message);              // "Привет"
  console.log(e.name);                 // "TypeError"
  console.log(e.fileName);             // "someFile.js"
  console.log(e.lineNumber);           // 10
  console.log(e.columnNumber);         // 0
  console.log(e.stack);                // "@Scratchpad/2:2:9\n"
}
</pre>

<h2 id="Specifications">Спецификации</h2>

{{Specifications}}

<h2 id="Browser_compatibility">Совместимость с браузерами</h2>

<div>{{Compat}}</div>

<h2 id="See_also">Смотрите также</h2>

<ul>
 <li>{{jsxref("Error")}}</li>
 <li>{{jsxref("TypeError.prototype")}}</li>
</ul>
